test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/AbstractCloudArchitectureIntegrationTest.java

```diff
@@ -0,0 +1,188 @@
+        assertThat(containersSpec).hasSize(containerIds.length);
+        List<String> containersSpecIds = getAllContainersSpecIds(containersSpec);
+        assertThat(containersSpecIds).containsExactlyInAnyOrder(containerIds);
+    }
+
+    private List<String> getAllContainersSpecIds(Collection<ContainerSpec> containerSpecs) {
+        List<String> ids = new ArrayList<>();
+        for (ContainerSpec containerSpec : containerSpecs) {
+            ids.add(containerSpec.getId());
+        }
+        return ids;
+    }
+
+    protected void verifyServerTemplateContainsKieServers(KieServerMgmtControllerClient kieControllerClient, String serverTemplate, int numberOfKieServers) {
+        Collection<ServerInstanceKey> kieServers = kieControllerClient.getServerTemplate(serverTemplate).getServerInstanceKeys();
+        assertThat(kieServers).hasSize(numberOfKieServers);
+    }
+
+    protected void verifySmartRouterContainsKieServers(KieServerRouterClient smartRouterAdminClient, int numberOfKieServers, List<String> containerIds, String... kieServerIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+
+        assertThat(routerConfig.getHostsPerServer()).containsKeys(kieServerIds);
+
+        for (String containerId : containerIds) {
+
+            // TODO: Commented until "Duplicate Kie server registration" issue gets resolved
+//            for (String kieServerId : kieServerIds) {
+//                assertThat(routerConfig.getHostsPerServer().get(kieServerId)).hasSize(numberOfKieServers);
+//            }
+            assertThat(routerConfig.getHostsPerContainer()).containsKey(containerId);
+            // TODO: Commented until "Duplicate Kie server registration" issue gets resolved
+//        assertThat(routerConfig.getHostsPerContainer().get(containerId)).hasSize(numberOfKieServers);
+            assertThat(routerConfig.getContainerInfosPerContainer()).containsKey(containerId);
+        }
+    }
+
+    protected void verifySmartRouterHostPerServerContainsContainers(KieServerRouterClient smartRouterAdminClient, String kieServerId, String... containersIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+        assertThat(routerConfig.getHostsPerServer().get(kieServerId)).contains(containersIds);
+    }
+
+    protected void verifySmartRouterHostPerContainerContainsServers(KieServerRouterClient smartRouterAdminClient, String containerId, String... kieServerIds) {
+        Configuration routerConfig = smartRouterAdminClient.getRouterConfig();
+        assertThat(routerConfig.getHostsPerContainer().get(containerId)).contains(kieServerIds);
+    }
+
+    protected void scaleKieServerTo(int count, KieServerDeployment... deployments) {
+        scaleKieServerTo(count, Arrays.asList(deployments));
+    }
+
+    protected void scaleKieServerTo(int count, List<KieServerDeployment> deployments) {
```

Comment:
Any reason for this method (and couple others) to not be `static`? What inner attributes of the class does it use or need?
Also, the order of its argument should, IMHO, be inverted. First say which deployments, then say how many.